grails-app/controllers/org/pih/warehouse/core/DocumentController.groovy

```diff
@@ -189,6 +193,10 @@ class DocumentController {
         def requestInstance = Requisition.get(command.requestId)
         def productInstance = Product.get(command.productId)
         def invoiceInstance = Invoice.get(command.invoiceId)
+        def stockMovement = outboundStockMovementService.getStockMovement(command.stockMovementId)
```

Comment:
Why def? :cry:

Reply:
because it can be either `OutboundStockMovement` or `StockMovement`

Reply:
essentially all you're checking here is that the provided `command.stockMovementId` is actually associated with a real stock movement right? All subsequent checks could just use `command.stockMovementId` if we wanted.

I get why you did it this way but I'm not the hugest fan of having to rope stock movement specific checks into the generic DocumentController. I know that's more a consequence of how this is all designed in the first place though so I don't expect you to change it. Maybe this could be refactored some day to have the command provide a single field that tells us what redirect it wants.

Reply:
@EWaterman generally, we have an access to `stockMovementInstance` from the document view model, but the problem is that if we want to eventually send it in the `uploadDocument` form as the `<g:hiddenField>`, it expects us to set a "raw" value (string, integer, boolean), so I could only set the id (string), so essentially we do the second call for the SM in this flow :man_facepalming: 
I don't like it as well, but yeah... this is GSP.....